dashbuilder/dashbuilder-backend/dashbuilder-services/src/main/java/org/dashbuilder/transfer/DataTransferServicesImpl.java

```diff
@@ -384,4 +415,70 @@ private File createVersionFile() throws java.io.IOException {
 
         return version;
     }
+
+    private List<Path> listPaths(FileSystem fs, Predicate<String> pathTester) {
+        List<Path> files = new ArrayList<>();
+        Path root = fs.getRootDirectories().iterator().next();
+        Files.walkFileTree(root, new SimpleFileVisitor<Path>() {
+
+            @Override
+            public FileVisitResult visitFile(Path path, BasicFileAttributes attrs) {
+                try {
+                    if (pathTester.test(path.toString())) {
+                        files.add(path);
+                    }
+                    return FileVisitResult.CONTINUE;
+
+                } catch (Exception e) {
+                    LOGGER.error(e.getMessage(), e);
+                    return FileVisitResult.TERMINATE;
+                }
+            }
+        });
```

Comment:
Consider using `DirectoryStream` here.

Reply:
I tried using it, The code was clearer:
~~~
        final Filter<Path> dirTest = p -> {
            return pathTester.test(p.toString());
        };
        try (DirectoryStream<Path> dirStream = Files.newDirectoryStream(root, dirTest)) {
            dirStream.forEach(files::add);
        } catch (Exception e) {
            throw new IllegalArgumentException("Error walking directory", e);
        }
~~~

but for some reason it was not going through some paths, specifically pages paths (plugin_content) hence it was not correctly helping me to filter the assets I wanted, so I went back to Files.walk